Patch by schwardt@univention.de (Thu Sep 10 11:34:08 2009 +0200):

turba: add objectclass filter to search criteria

While searching turba uses either an objectclass filter or user defined
criteria but not both. When searching for "name" all ldap object with
matching "cn" attribute will show up. Selecting one returns a "no result"
message.
This patch adds the objectclass filter if definded, so only valid address
book entries are shown.


--- a/horde-webmailer/turba/lib/Driver/ldap.php
+++ b/horde-webmailer/turba/lib/Driver/ldap.php
@@ -155,6 +155,10 @@ class Turba_Driver_ldap extends Turba_Driver {
                     $filter .= '(&' . $this->_buildSearchQuery($vals) . ')';
                 }
             }
+            $objfilter = $this->_buildObjectclassFilter();
+            if (!empty($objfilter)) {
+                $filter = '(&' . $filter . $objfilter . ')';
+            }
         } else {
             /* Filter on objectclass. */
             $filter = $this->_buildObjectclassFilter();
